### ISSUE TITLE
tp::maintain_order is unattainable in the general case, and fails in a specific case

### ISSUE THREAD

**Issue by Mortal** [thomasmoelhave/tpie#141]
It is not possible to maintain the order in parallel in the general case when the output items do not correspond one-to-one to the input items.

The current implementation fails to maintain the order when the nodes do their own buffering even when correspondence is one-to-one. Perhaps we should support maintaining the order in this case.

Or perhaps we should just note that maintain_order won't do what the user expects if they do their own buffering.
